.github/scripts/compile.sh

```diff
@@ -35,13 +35,37 @@ do_patch () {
 
 WORKING=`pwd`
 wget ${PROTON_SOURCE_URL} -O qpid-proton.tar.gz
+wget ${LWS_SOURCE_URL} -O libwebsockets.tar.gz
 
-mkdir -p qpid-proton-src build staging proton_build proton_install
+mkdir -p qpid-proton-src build staging proton_build proton_install lws-src lws_build lws_install
 tar -zxf qpid-proton.tar.gz -C qpid-proton-src --strip-components 1
+tar -zxf libwebsockets.tar.gz -C lws-src --strip-components 1
 
 do_patch "patches/proton" qpid-proton-src
 
-cd proton_build
+cd $WORKING/lws_build
+cmake \
+    -D LWS_LINK_TESTAPPS_DYNAMIC=ON \
+    -D LWS_WITH_LIBUV=OFF \
+    -D LWS_WITHOUT_BUILTIN_GETIFADDRS=ON \
+    -D LWS_USE_BUNDLED_ZLIB=OFF \
+    -D LWS_WITHOUT_BUILTIN_SHA1=ON \
+    -D LWS_WITH_STATIC=OFF \
+    -D LWS_IPV6=ON \
+    -D LWS_WITH_HTTP2=OFF \
+    -D LWS_WITHOUT_CLIENT=OFF \
+    -D LWS_WITHOUT_SERVER=OFF \
+    -D LWS_WITHOUT_TESTAPPS=ON \
+    -D LWS_WITHOUT_TEST_SERVER=ON \
+    -D LWS_WITHOUT_TEST_SERVER_EXTPOLL=ON \
+    -D LWS_WITHOUT_TEST_PING=ON \
+    -D LWS_WITHOUT_TEST_CLIENT=ON \
```

Comment:
What is the logic for choosing what options to enable? Is it a list of all switches currently available, or is it carefully chosen?

Reply:
I am guessing these options were chosen based off off the build log of LWS in brew.

Reply:
Could be, the Rawhide version is https://src.fedoraproject.org/rpms/libwebsockets/blob/rawhide/f/libwebsockets.spec#_58

Reply:
Looking at that rawhide link, it seems to me that the `LWS_WITH_LIBUV` option by default is `OFF`
You can see that in the LWS CMakeLists.txt
https://github.com/warmcat/libwebsockets/blob/main/CMakeLists.txt#L215

But in brew, this option is set to ON

Reply:
I used what was here https://dist-git.host.prod.eng.bos.redhat.com/cgit/rpms/libwebsockets/ as a starting point and then turned LIBUV off

Reply:
makes sense, for lws 3.x, I think; for latest lws the reasonable defaults would be the fedora rawhide